### ISSUE TITLE
Github releases may not be the latest

### ISSUE THREAD

**Issue by yellowhat** [Mic92/nix-update#272]
Hi,

some projects may release new minor/patch releases after a new major/minor.

For example, at the time of writing https://github.com/kubernetes/kubernetes/tags:

* v1.30.4
* v1.29.8
* v1.28.13
* v1.31.0
 
therefore:

```console
$ nix --extra-experimental-features "flakes nix-command" run github:Mic92/nix-update -- --flake kubernetes
...
trying to fetch https://api.github.com/repos/kubernetes/kubernetes/releases
Not updating version, already 1.30.4
```

`nix-update` assumes that `1.30.4` is the latest but it is not.

Any suggestions?

Thanks

**Comment by adamcstephens** [comment]
Is this a duplicate of #271 ?

**Comment by jian-lin** [comment]
In 1.5.0, `nix-update` checks releases instead of tags. `v1.31.0` is one of the [tags] but not one of the [releases].

related: https://github.com/Mic92/nix-update/pull/246

[tags]: https://github.com/kubernetes/kubernetes/tags
[releases]: https://github.com/kubernetes/kubernetes/releases

**Comment by yellowhat** [comment]
> In 1.5.0, `nix-update` checks releases instead of tags. `v1.31.0` is one of the [tags](https://github.com/kubernetes/kubernetes/tags) but not one of the [releases](https://github.com/kubernetes/kubernetes/releases).
> 
> related: #246

1.31.0 is a release and marked as latest

![image](https://github.com/user-attachments/assets/79335b4b-7a6d-464f-b5a3-97775627a35c)

**Comment by lilyinstarlight** [comment]
#246 and #269 should be reverted as mentioned in <https://github.com/Mic92/nix-update/pull/246#issuecomment-2323018854>. That change did not actually fix any package updates (since version-regex was already the tool to help the listed motivation), but it did break many packages like this one

I've just forked and reverted myself at <https://github.com/lilyinstarlight/nix-update> for now so I can have an actually working nix-update in the mean time

**Comment by lilyinstarlight** [comment]
Fixed in #274
